### PR TITLE
20 notifications of events

### PR DESCRIPTION
**Pull request by lyeskara**

### CODE REVIEW

**Review by kalsij — APPROVED** [review]
Looks good seems like notifications should be working!

**Comment by kalsij** [comment]
effects #292

File: frontend/src/components/DirectMessage/Messaging.js [hunk 1]
```diff
@@ -74,47 +89,76 @@ const Message = () => {
...
+          let condition = false
+          notifications_array.forEach((notif) => {
+            if (!(notif.receipient_notifications === note.receipient_notifications)) {
+              condition = true;
+            }
+          })
+          if (condition) {
+            notifications_array.push(note)
+          }
+          updateDoc(doc(collection(db, 'Notifications'), recipientId), {
```
Comment (Chovv) [inline frontend/src/components/DirectMessage/Messaging.js:157]: I see you're also storing the notifications in the DB.

File: frontend/src/components/UserFeedPage/CreatPost.js [hunk 2]
```diff
@@ -91,31 +92,36 @@ function CreatPost() {
...
-                let condition = false
-                notifications_array.forEach((notif) => {
-                  if (!(notif.id === note.id)) {
-                    condition = true;
-                  }
-                })
-                if (condition) {
-                  notifications_array.push(note)
+            getDoc(doc(collection(db, "notification_settings"), id)).then((note_data) => {
+              if (note_data.data() !== undefined) {
```
Comment (Chovv) [inline frontend/src/components/UserFeedPage/CreatPost.js:96]: I see you made it more robust by checking if there's an undefined value. Good thinking.

**Review by Chovv — COMMENTED** [review]
Looks good to me, could use more documentation for clarity sake though.

**Pull request merged**